index.js

```diff
@@ -30,6 +31,27 @@ function deploy(workingDir, baseURL, branchOrPack, config, cb) {
   if (shell.test('-f', path.resolve(branchOrPack))) {
     return httpPutDeployment(baseURL, config, branchOrPack, cb);
   } else {
+    if (!branchOrPack || branchOrPack === 'deploy') {
+      branchOrPack = defaultPackagePath(workingDir);
```

Comment:
if `branchOrPack` started out as `somebranch` referring to the branch and, this statement would reset it to null. Subsequently, `gitDeploy` would attempt to deploy from the wrong branch